### DISCUSSION TITLE
Help with ArgoCD Disaster Recovery

### DISCUSSION THREAD

**Discussion by AlessioCasolaro** [argoproj/argo-cd#19769] [Q&A]
Hello, I am exploring ArgoCD's disaster recovery capabilities and came across the backup system mentioned in the documentation: [](https://argo-cd.readthedocs.io/en/stable/operator-manual/disaster_recovery/)
However, I would like to understand if there is an alternative approach I could take.

Let's assume I have a Volume that already has disaster recovery measures in place. I am using this Volume to store the Redis instance connected to ArgoCD. In the event of a disaster, could I simply attach a new ArgoCD deployment to the existing Volume? 
Would I find all of ArgoCD's configurations and applications intact? Or would I lose some information/configurations, thus necessitating the use of the backup command suggested in the guide?

I would greatly appreciate it if you could provide some guidance on this matter. If there are any limitations or caveats to using the approach described.

Thank you in advance for your assistance.

**Comment by rumstead** [comment]
Here are some related discussions on DR.
1. https://github.com/argoproj/argo-cd/discussions/9488
1. https://github.com/argoproj/argo-cd/discussions/13126

**Reply by AlessioCasolaro** [reply]
Perhaps the 2 discussion could partially answer my doubt, although my question was more specific. Argo being stateless, the only one remaining is Redis which has its own volume, could I potentially hook a new deployment of Argo (Redis to new included) by hooking into the volume of the previous Redis?
Would I have the data at that point?

**Reply by rumstead** [reply]
Redis is just used to store rendered git manifests. It won't have the Kubernetes resources that configure Argo CD and are stored in etcd, eg Applications, config maps, and secrets. You must also deploy those in the new cluster housing Argo CD during a DR event.

**Reply by AlessioCasolaro** [reply]
OK, clear. Now I wonder though, Argo being stateless, where the configurations (applications, secrets, etc.) are actually saved.

**Reply by AlessioCasolaro** [reply]
@rumstead I found this answer [in one of the questions](https://github.com/argoproj/argo-cd/issues/4255#issuecomment-686916476), the most of the work seems to be delegated to etcd and CRDs,  so technically if those stay up, I could safely throw out my Argo deployment and recreate it finding all the info regarding Applications is that correct?

**Reply by rumstead** [reply]
Yes, all the Kubernetes resources configuring Argo CD live in ectd, and all Kubernetes resources live there. What kind of disaster recovery are you trying to solve for? Typically, you want a story around the Kubernetes cluster going down, including ectd.
